src/lib.rs

```diff
@@ -0,0 +1,324 @@
+use crate::exec::{
+    is_addr_in_exec, is_common_parent, AtomicExec, AtomicExecParamsRaw, ExecStatus, LockedOutput,
+    SubmitExecParams, SubmitOutput,
+};
+use crate::state::State;
+use cid::Cid;
+use fil_actors_runtime::runtime::{ActorCode, Runtime};
+use fil_actors_runtime::{actor_error, cbor, ActorDowncast, ActorError, INIT_ACTOR_ADDR};
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_encoding::RawBytes;
+use fvm_shared::error::ExitCode;
+use fvm_shared::{MethodNum, METHOD_CONSTRUCTOR};
+use num_derive::FromPrimitive;
+use num_traits::FromPrimitive;
+
+use crate::subnet_id::SubnetID;
+use crate::types::ConstructorParams;
+
+mod address;
+mod atomic;
+mod cross;
+mod error;
+mod exec;
+mod state;
+mod subnet_id;
+mod types;
```

Comment:
Most of those modules are just copied from IPC GW over. We certainly need to apply the DRY principle here.

Reply:
I created the issue already

Reply:
I think this is fixed!